src/Symfony/Component/Mailer/Event/MessageEvent.php

```diff
@@ -64,4 +69,24 @@ public function isQueued(): bool
     {
         return $this->queued;
     }
+
+    public function addStamp(StampInterface $stamp): void
+    {
+        if (!$this->queued) {
+            throw new LogicException(sprintf('Cannot call `%s()` on a message that is not meant to be queued. Set $queued to `true` in the constructor.', __METHOD__));
```

Comment:
I would remove the second part which does not seem relevant for end user code.

Reply:
done